src/lib/pubkey/classic_mceliece/cmce_poly.cpp

```diff
@@ -0,0 +1,175 @@
+ *
+ * @param bytes input bytes
+ * @return The vector of GF_Elem elements
+ */
+std::vector<GF_Elem> load_le_gf_vec(std::span<const uint8_t> bytes) {
+   std::vector<GF_Elem> result;
+   BufferSlicer bs(bytes);
+   result.reserve(bytes.size() / sizeof(GF_Elem));
+   for(size_t i = 0; i < bytes.size() / sizeof(GF_Elem); ++i) {
+      result.emplace_back(load_le<uint16_t>(bs.take<sizeof(GF_Elem)>()));
+   }
+   return result;
+}
+
+}  // namespace
+
+Classic_McEliece_GF Classic_McEliece_Polynomial::operator()(Classic_McEliece_GF a) const {
+   BOTAN_ASSERT(a.modulus() == coef_at(0).modulus(), "Unmatching Galois fields");
+
+   Classic_McEliece_GF r(GF_Elem(0), a.modulus());
+   for(auto it = m_coef.rbegin(); it != m_coef.rend(); ++it) {
+      r *= a;
+      r += *it;
+   }
+
+   return r;
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::multiply(const Classic_McEliece_Polynomial& a,
+                                                                       const Classic_McEliece_Polynomial& b) const {
+   std::vector<Classic_McEliece_GF> prod(m_t * 2 - 1, {GF_Elem(0), m_poly_f});
+
+   for(size_t i = 0; i < m_t; ++i) {
+      for(size_t j = 0; j < m_t; ++j) {
+         prod.at(i + j) += (a.coef_at(i) * b.coef_at(j));
+      }
+   }
+
+   for(size_t i = (m_t - 1) * 2; i >= m_t; --i) {
+      for(auto& [idx, coef] : m_position_map) {
+         prod.at(i - m_t + idx) += coef * prod.at(i);
+      }
+   }
+
+   prod.erase(prod.begin() + m_t, prod.end());
+
+   return Classic_McEliece_Polynomial(prod);
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::create_element_from_bytes(
+   std::span<const uint8_t> bytes) const {
+   BOTAN_ARG_CHECK(bytes.size() == m_t * 2, "Correct input size");
+   auto coef = load_le_gf_vec(bytes);
+   return create_element_from_coef(coef);
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::create_element_from_coef(
+   const std::vector<GF_Elem>& coeff_vec) const {
+   std::vector<Classic_McEliece_GF> coeff_vec_gf;
+   std::transform(coeff_vec.begin(), coeff_vec.end(), std::back_inserter(coeff_vec_gf), [this](auto& coeff) {
+      return Classic_McEliece_GF(coeff, m_poly_f);
+   });
+   return Classic_McEliece_Polynomial(coeff_vec_gf);
+}
+
+bool operator==(const Classic_McEliece_Polynomial_Ring::Big_F_Coefficient& lhs,
+                const Classic_McEliece_Polynomial_Ring::Big_F_Coefficient& rhs) {
+   return lhs.coeff == rhs.coeff && lhs.idx == rhs.idx;
+}
+
+std::optional<Classic_McEliece_Minimal_Polynomial> Classic_McEliece_Polynomial_Ring::compute_minimal_polynomial(
+   std::span<const uint8_t> seed) const {
+   auto polynomial = create_element_from_bytes(seed);
+   std::vector<Classic_McEliece_Polynomial> mat;
+
+   mat.push_back(create_element_from_coef(concat_as<std::vector<GF_Elem>>(
+      std::vector<GF_Elem>{GF_Elem(1)}, std::vector<GF_Elem>(degree() - 1, GF_Elem(0)))));
+
+   mat.emplace_back(polynomial);
```

Comment:
```suggestion
   mat.push_back(polynomial);
```

That's a nit. Though, `emplace_back()` doesn't have a benefit in this case. You're copying the `polynomial` anyway. Typically, `emplace_back()` makes sense when you want to call the constructor of the contained object in the final container location. Here, you basically call the copy constructor in-place.